src/WebUI/Components/DaemonSetListingComponent.tsx

```diff
@@ -26,97 +27,94 @@ export class DaemonSetListingComponent extends BaseComponent<IDaemonSetComponent
-                }
-                return (
-                    <div>
-                        {
-                            !!statusProps &&
-                            <Status {...statusProps} size={StatusSize.m} />
-                        }
-                        <span>{podString}</span>
-                    </div>);
-            }
-            case ageKey: {
-                return (<Ago date={new Date(daemonSet.metadata.creationTimestamp)} />);
-            }
+    private static _renderSetNameCell(rowIndex: number, columnIndex: number, tableColumn: ITableColumn<V1DaemonSet>, daemonSet: V1DaemonSet): JSX.Element {
+        return (
+            <SimpleTableCell columnIndex={columnIndex} tableColumn={tableColumn} key={"col-" + columnIndex}>
+                {ListComponent.renderTwoLineColumn(daemonSet.metadata.name,
+                    Utils.getPipelineText(daemonSet.metadata.annotations),
+                    colDataClassName, "primary-text", "secondary-text")}
+            </SimpleTableCell>);
+    }
+
+    private static _renderImageCell(rowIndex: number, columnIndex: number, tableColumn: ITableColumn<V1DaemonSet>, daemonSet: V1DaemonSet): JSX.Element {
+        const textToRender = daemonSet.spec.template.spec.containers[0].image;
+        return (
+            <SimpleTableCell columnIndex={columnIndex} tableColumn={tableColumn} key={"col-" + columnIndex}>
+                {ListComponent.renderColumn(textToRender || "", ListComponent.defaultColumnRenderer, colDataClassName)}
+            </SimpleTableCell>);
+    }
+
+    private static _renderPodsCell(rowIndex: number, columnIndex: number, tableColumn: ITableColumn<V1DaemonSet>, daemonSet: V1DaemonSet): JSX.Element {
+        let statusProps: IStatusProps | undefined;
+        let podString: string = "";
+        if (daemonSet.status.desiredNumberScheduled > 0) {
+            statusProps = Utils._getPodsStatusProps(daemonSet.status.currentNumberScheduled, daemonSet.status.desiredNumberScheduled);
+            podString = format("{0}/{1}", daemonSet.status.currentNumberScheduled, daemonSet.status.desiredNumberScheduled);
         }
-        return ListComponent.renderColumn(textToRender||"",ListComponent.defaultColumnRenderer,colDataClassName);
+
+        return (
+            <SimpleTableCell columnIndex={columnIndex} tableColumn={tableColumn} key={"col-" + columnIndex}>
+                <div>
+                    {
+                        !!statusProps &&
+                        <Status {...statusProps} size={StatusSize.m} />
+                    }
+                    <span>{podString}</span>
+                </div>
+            </SimpleTableCell>);
+    }
+
+    private static _renderAgeCell(rowIndex: number, columnIndex: number, tableColumn: ITableColumn<V1DaemonSet>, daemonSet: V1DaemonSet): JSX.Element {
+        return (
+            <SimpleTableCell columnIndex={columnIndex} tableColumn={tableColumn} key={"col-" + columnIndex}>
```

Comment:
would it be good to have a function or a component takes the inner html and component is rendered,

<SimpleTableCell columnIndex={columnIndex} tableColumn={tableColumn} key={"col-" + columnIndex}>
                {{here_we_will_have_inner_component}}
            </SimpleTableCell>

Reply:
Added a helper function renderTableCell to ListComponent and using that in all components